types.d.ts

```diff
@@ -154,8 +154,8 @@ type ActionCreatorOrString4<Arg1, Arg2, Arg3, Arg4, P, M={}> = ActionCreator4<Ar
 type ActionCreatorOrString5<Arg1, Arg2, Arg3, Arg4, Arg5, P, M={}> = ActionCreator5<Arg1, Arg2, Arg3, Arg4, Arg5, P, M> | string
 type ActionCreatorOrString6<Arg1, Arg2, Arg3, Arg4, Arg5, Arg6, P, M={}> = ActionCreator6<Arg1, Arg2, Arg3, Arg4, Arg5, Arg6, P, M> | string
 
-interface Reducer<S> {
-  (state: S, action: Redux.AnyAction): S
+interface Reducer<S, A extends Redux.Action = Redux.AnyAction> {
+  (state: S | undefined, action: A): S
```

Comment:
When would the `state` be `undefined` in the reduce function?

Reply:
Just for compatibility with [redux Reducer type](https://github.com/reduxjs/redux/blob/master/index.d.ts#L62)

It was updated on this PR: https://github.com/reduxjs/redux/pull/2563